tensorflow/core/kernels/resize_bicubic_op.cc

```diff
@@ -330,48 +331,62 @@ inline void interpolate_with_caching(
                       x_wai.weight_2, x_wai.weight_3);
         }
       } else {
-        for (int64 c = 0; c < num_channels; ++c) {
-          float cached_value[4] = {0};
-          for (int64 x = 0; x < resizer_state.out_width; ++x) {
-            const WeightsAndIndices& x_wai = x_wais[x];
-            // Shift values in cached_value to fill first 'advance' values.
-            switch (x_wai.advance) {
-              case 3:
-                cached_value[0] = cached_value[1];
-                cached_value[1] = cached_value[2];
-                cached_value[2] = cached_value[3];
-                break;
-              case 2:
-                cached_value[0] = cached_value[2];
-                cached_value[1] = cached_value[3];
-                break;
-              case 1: {
-                cached_value[0] = cached_value[3];
-                break;
+        std::vector<float> cached_value(4 * num_channels, 0);
```

Comment:
this can be declared outside all loops (I guess even declare on the 3-channel case, or in the 3-channel case declare it of size 0), to save an allocation per row.

Reply:
Thanks @cwhipkey. That indeed will help avoid allocation. Updated.